#PANDAS
## 1. Pandas series is a one dimensional array

In [6]:
import pandas as pd

ageOfStudents=pd.Series (
 [23, 13, 56, 34, 12, 41],
 index = ["Jane", "Joseph", "Simon", "Joan", "Peter", "James"], 
 name = "Age Of Students"  
)
ageOfStudents

Jane      23
Joseph    13
Simon     56
Joan      34
Peter     12
James     41
Name: Age Of Students, dtype: int64

### A DataFrame is a Two Dimensional Array.


This is to mean that it contains rows and columns.

In [54]:
students_data = {
    "Name":["Jane","Joseph","Simon","Joan","Peter","James"],
    "Maths":[63, 73, 56, 34, 92, 41],
    "Age":[23, 21, 19, 24, 22, 21],
    "English":[77, 87, 66, 90, 45, 76],

}
import pandas as pd
student =pd. DataFrame( students_data) 
student


,Name,Maths,Age,English
0,Jane,63,23,77
1,Joseph,73,21,87
2,Simon,56,19,66
3,Joan,34,24,90
4,Peter,92,22,45
5,James,41,21,76


In [ ]:
student["Age"]

0    23
1    21
2    19
3    24
4    22
5    21
Name: Age, dtype: int64

1. What is the modal age

In [ ]:
student["Age"].mode() [0] 

np.int64(21)

From the results above, the modal age of students is 21.

In [ ]:
student["Maths"].mode ()

0    34
1    41
2    56
3    63
4    73
5    92
Name: Maths, dtype: int64

2. What is the mean mark of maths?

In [ ]:
student["Maths"].mean ()

np.float64(59.833333333333336)

From the results above, the modal mark of Maths is 59.8334

3. What is the variance of Maths?

In [ ]:
student ["Maths"]. var ()

np.float64(450.9666666666666)

Fromthe results above, the variance of Maths is 450.9667

4. What is the standard deviation of maths?

In [ ]:
student ["Maths"]. std()

np.float64(21.235975764411357)

From the above results, the Standard Variation is 21.2360

5. What is the total_marks of both English and Maths?

In [67]:
Total_marks= student[['English','Maths']].sum().sum()
print("Total_marks:", Total_marks)

Total_marks: 800


The total marks of both English and math is 800

6. Find the average of the total marks.

In [66]:
Average_mark = Total_marks/(len(student)*2)
print("Average_mark:", Average_mark)

Average_mark: 66.66666666666667


The average of the total marks is 66.6667

7. What is the modal mark of English?

In [ ]:
student ["English"]. mode ()

0    45
1    66
2    76
3    77
4    87
5    90
Name: English, dtype: int64

8. What is mean mark of English?

In [ ]:
student ["English"].mean ()

np.float64(73.5)

The mean mark of Enlish is 73.5

9. What is the Standard deviation for English?

In [ ]:
student ["English"]. std()

np.float64(16.37986568931504)

The standard deviation for English is 16.3799.

10. What is the range of the age values in our data?

In [ ]:
range = student ["Age"].max()-student ["Age"].min()
range

np.int64(5)

The range of the Age values is 5.

11. Which student scored the highest mark in English?

In [ ]:
student ["Name"] [student ["English"].idxmax ()]

'Joan'

Joan scored the highest mark in English.